inspire_records/api/base.py

```diff
@@ -24,8 +24,62 @@
 
 from __future__ import absolute_import, division, print_function
 
+from abc import ABCMeta, abstractmethod
+import uuid
+
+from invenio_pidstore.models import PersistentIdentifier
+from inspire_dojson.utils import strip_empty_values
+from inspire_schemas.api import validate
+from invenio_pidstore.errors import PIDDoesNotExistError
 from invenio_records_files.api import Record
+from invenio_db import db
 
 
 class InspireRecord(Record):
     """Inspire Record."""
+
+    pid_type = None
+
+    @staticmethod
+    def get_uuid_from_pid_value(pid_type, pid_value):
+        pid = PersistentIdentifier.get(pid_type, pid_value)
+        return pid.object_uuid
+
+    @staticmethod
+    def strip_empty_values(data):
+        return strip_empty_values(data)
+
+    @staticmethod
+    def mint(record_uuid, data):
+        pass
+
+    def validate(self):
+        validate(self)
+
+    @classmethod
+    def get_record_by_pid_value(cls, pid_value):
+        record_uuid = cls.get_uuid_from_pid_value(cls.pid_type, pid_value)
+        record = cls.get_record(record_uuid)
+        return cls(record.json, model=record)
+
+    @classmethod
+    def create(cls, data, id_=None, **kwargs):
+        id_ = id_ or uuid.uuid4()
```

Comment:
Check if this ID already exists?

Reply:
> Check if this ID already exists?
This will throw an exception anyhow, an extra query just for checking IMO doesn't worth
> Can you describe use case when we want to pass id_ by ourself?
https://github.com/inveniosoftware/invenio-records/blob/master/invenio_records/api.py#L141